packages/account/src/providers/provider.ts

```diff
@@ -379,15 +379,28 @@ type ChainInfoCache = Record<string, ChainInfo>;
  */
 type NodeInfoCache = Record<string, NodeInfo>;
 
-type Operations = ReturnType<typeof getOperationsSdk>;
-
-type SdkOperations = Omit<Operations, 'submitAndAwait' | 'statusChange'> & {
-  submitAndAwait: (
-    ...args: Parameters<Operations['submitAndAwait']>
-  ) => Promise<ReturnType<Operations['submitAndAwait']>>;
-  statusChange: (
-    ...args: Parameters<Operations['statusChange']>
-  ) => Promise<ReturnType<Operations['statusChange']>>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PromisifyReturn<OfFn extends (...args: any) => any> = (
+  ...args: Parameters<OfFn>
+) => Promise<ReturnType<OfFn>>;
+
+type Ops = ReturnType<typeof getOperationsSdk>;
+
+type Operations = {
+  [K in keyof Ops]: ReturnType<Ops[K]> extends AsyncIterable<unknown>
+    ? PromisifyReturn<Ops[K]> // Wrap subscriptions in a promise to match return of `FuelGraphqlSubscriber`
+    : Ops[K];
+};
```

Comment:
Sure, we can do it like that, although there are now two `ReturnType<typeof getOperationsSdk>` calls which both calculate the types. It probably won't matter in the grand scheme of things, but I try to keep type performance in mind to avoid death by a thousand cuts. This is a nice read: https://github.com/microsoft/TypeScript/wiki/Performance. Feel free to commit your suggestion or resolve the conversation if you change your mind.